frontend/src/components/forms/CreateResident.tsx

```diff
@@ -0,0 +1,254 @@
+
+    const handleBuildingChange = (
+        selectedOption: SingleValue<{ label: string; value: string }>,
+    ) => {
+        if (selectedOption !== null) {
+            setBuilding(selectedOption.value);
+            setBuildingError(false);
+        } else {
+            setBuildingError(true);
+        }
+    };
+
+    const handleOpen = () => {
+        setIsOpen(true);
+
+        // Reset the input states
+        setInitials("");
+        setRoomNumber("");
+        setMoveInDate(new Date());
+        setBuilding("");
+
+        // Reset the error states
+        setInitialsError(false);
+        setRoomNumberError(false);
+        setMoveInDateError(false);
+        setBuildingError(false);
+
+        //  Reset alert state
+        setShowAlert(false);
+    };
+
+    const handleClose = () => {
+        setIsOpen(false);
+    };
+
+    const handleSubmit = () => {
+        setInitialsError(initials === "");
+        setRoomNumberError(roomNumber === "");
+        setBuildingError(building === "");
+
+        //  Prevents form submission if any required values are missing
+        if (initials === "" || roomNumber === "" || building === "") {
+            return;
+        }
+
+        addResident();
+
+        setIsOpen(false);
+        setShowAlert(true);
+    };
+
+    // Timer to remove alert
+    useEffect(() => {
+        if (showAlert) {
+            setTimeout(() => {
+                setShowAlert(false);
+            }, 3000)
+        }
+    }, [showAlert]);
+
+    return (
+        <>
+            <Box textAlign="right">
+                <Button
+                    onClick={handleOpen}
+                    marginBottom="16px"
+                    variant="primary"
+                >
+                    <AddIcon boxSize="16px" marginRight="8px" />
+                    Add Resident
+                </Button>
+            </Box>
+
+            <Box>
+                <Modal isOpen={isOpen} onClose={handleClose} size="xl">
+                    <ModalOverlay />
+                    <ModalContent>
+                        <ModalHeader>Add Resident</ModalHeader>
+                        <ModalCloseButton size="lg" />
+                        <ModalBody>
+                            <Divider />
+                            <Row style={{ marginTop: "16px" }}>
+                                <Col>
+                                    <FormControl isRequired isInvalid={initialsError}>
+                                        <FormLabel>Resident Initials</FormLabel>
+                                        <Input
+                                            placeholder="AA"
+                                            value={initials}
+                                            onChange={handleInitialsChange}
+                                        />
+                                    </FormControl>
+                                    <FormErrorMessage>Resident Initials are required and must contain 2 letters.</FormErrorMessage>
```

Comment:
same as below: You need to add this in the `FormControl` component, so move it up one line